packages/components/addon/components/hds/tabs/index.js

```diff
@@ -0,0 +1,90 @@
+import Component from '@glimmer/component';
+import { A } from '@ember/array';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class HdsTabsIndexComponent extends Component {
+  // UNCOMMENT THIS IF YOU NEED A CONSTRUCTOR
+  // constructor() {
+  //   super(...arguments);
+  //   // ADD YOUR ASSERTIONS HERE
+  // }
+
+  @tracked tabNodes = A([]);
+  @tracked tabIds = A([]);
+  @tracked panelNodes = A([]);
+  @tracked panelIds = A([]);
+  @tracked selectedIndex = 0; // isSelected, 0 = default
+
+  @action
+  didInsertTab(tabId, isSelected, element) {
+    this.tabNodes = A([...this.tabNodes, element]);
+    this.tabIds = A([...this.tabIds, tabId]);
+
+    if (isSelected) {
+      this.setSelectedTabIndex(this.tabIds.length - 1);
+    }
+  }
+
+  @action
+  didInsertPanel(panelId, element) {
+    this.panelNodes = A([...this.panelNodes, element]);
+    this.panelIds = A([...this.panelIds, panelId]);
+  }
+
+  @action
+  handleClick(index, e) {
+    this.setSelected(index, e);
+  }
+
+  @action
+  handleKeyUp(index, e) {
+    const leftArrow = 37;
+    const rightArrow = 39;
+    const downArrow = 40;
+
+    if (e.keyCode === rightArrow) {
+      if (index < this.tabIds.length - 1) {
+        index++;
+        this.setSelected(index, e);
+      }
+    } else if (e.keyCode === leftArrow) {
+      if (index > 0) {
+        index--;
+        this.setSelected(index, e);
+      }
+    } else if (e.keyCode === downArrow) {
+      e.preventDefault();
+      this.setSelectedPanelFocus(index, e);
+    }
+  }
+
+  setSelected(index, e) {
+    e.preventDefault();
+    this.selectedIndex = index;
+    this.tabNodes[index].focus();
+  }
+
+  setSelectedTabIndex(index) {
+    this.selectedIndex = index;
+    this.tabNodes[index].focus();
+  }
```

Comment:
When reading the code, I've found these two functions with similar names confusing (to me). 
Maybe we could have only `setSelectedTabIndex` and whenever the `setSelected` is used, have this:
```
e.preventDefault();
this.setSelectedTabIndex(index);
```

in this way there would be only one function that "sets" the selected tab index and one would not have doubt which one to use, in which context

Reply:
Thanks, I attempted to clarify these. Please open another comment if you think it still needs work.